src/pokemon/pokemon_storage_system.c

```diff
@@ -1,93 +1,711 @@
+            return i;
+    }
+    return -1;
+}
 
-void sub_8098898(u8 index) {
-    u8 *ptr;
+u8 GetNumValidDaycarePartyMons(void)
+{
+    u16 i;
+    u16 count;
 
-    Menu_DrawStdWindowFrame(10, 16, 29, 19);
+    for (i = 0, count = 0; i < PARTY_SIZE; i++)
+    {
+        struct Pokemon *pokemon = gPlayerParty + i;
+        if (GetMonData(pokemon, MON_DATA_SPECIES) != 0 && !GetMonData(pokemon, MON_DATA_IS_EGG))
+            count++;
+    }
+    return count;
+}
+
+u8 CountAlivePartyMonsExceptOne(u8 toSkip)
+{
+    u16 i;
+    u16 count;
 
-    switch (gUnknown_083B6DF4[index].format)
+    for (i = 0, count = 0; i < PARTY_SIZE; i++)
     {
+        if (i != toSkip)
+        {
+            struct Pokemon *pokemon = gPlayerParty + i;
+            if (GetMonData(pokemon, MON_DATA_SPECIES) != 0 && !GetMonData(pokemon, MON_DATA_IS_EGG) && GetMonData(pokemon, MON_DATA_HP) != 0)
+                count++;
+        }
+    }
+    return count;
+}
+
+u8 CountAlivePartyMonsExceptSelectedOne(void)
+{
+    return CountAlivePartyMonsExceptOne(gSpecialVar_0x8004);
+}
 
-    case 2:
-        ptr = StringCopy(gUnk2002694, gUnknown_083B6DF4[index].text);
-        ptr = StringCopy(ptr, gUnk20011fa);
-        break;
+u8 StorageSystemGetPartySize(void)
+{
+    u16 i;
+    u16 count;
 
-    case 5:
-        ptr = StringCopy(gUnk2002694, gUnknown_083B6DF4[index].text);
-        ptr = StringCopy(ptr, gUnk20026e4);
-        break;
+    for (i = 0, count = 0; i < PARTY_SIZE; i++)
+    {
+        if (GetMonData(gPlayerParty + i, MON_DATA_SPECIES) != 0)
+            count++;
+    }
+    return count;
+}
+
+u8 *unref_sub_8095C60(u8 *dest, const u8 *src, u16 pad)
+{
+    u8 *_dest = StringCopy(dest, src);
+    while (_dest < dest + pad)
+    {
+        *_dest++ = CHAR_SPACE;
+    }
+    *_dest = EOS;
+    return _dest;
+}
 
-    case 1:
-        // {var} + " is selected."
-        ptr = StringCopy(gUnk2002694, gUnk20011fa);
-        ptr = StringCopy(ptr, gUnknown_083B6DF4[index].text);
-        break;
+void sub_8095C8C(u16 *dest, u16 dest_left, u16 dest_top, const u16 *src, u16 src_left, u16 src_top, u16 dest_width, u16 dest_height, u16 src_width)
+{
+    u16 i;
 
-    case 4:
-        // {var} + " was released."
-        ptr = StringCopy(gUnk2002694, gUnk20026e4);
-#if ENGLISH
-        ptr = StringCopy(ptr, gUnknown_083B6DF4[index].text);
-#elif GERMAN
-        ptr = de_sub_8073174(gUnk2002694, gUnknown_083B6DF4[index].text);
+    dest_width *= 2;
+    dest += dest_top * 0x20 + dest_left;
+    src += src_top * src_width + src_left;
+    for (i = 0; i < dest_height; i++)
+    {
+        CpuCopy16(src, dest, dest_width);
+        dest += 0x20;
+        src += src_width;
+    }
+}
+
+#ifdef NONMATCHING
+void unref_sub_8095D08(u16 *dest, u16 dest_left, u16 dest_top, u16 width, u16 height)
+{
+    u16 i;
+
+    dest += dest_top * 0x20 + dest_left;
+    width *= 2;
+    for (i = 0; i < height; dest += 0x20, i++)
```

Comment:
This looks similar to the type of modified DmaLoop that ProcessDma3Requests uses where it checks size (first) before the loop. Have you tried using that macro instead?